code/modules/projectiles/guns/energy/special.dm

```diff
@@ -152,15 +156,29 @@
 		..()
 
 // Tool procs, in case plasma cutter is used as welder
+// Plasma cutters do not toggle so they need to explicitly use charge
+/obj/item/gun/energy/plasmacutter/tool_start_check(mob/living/user, amount=0)
+	if(!use(amount)) //nearly everything calls with amount=0 explicitly, handed in use()

```

Comment:
https://github.com/tgstation/tgstation/blob/76ffc246cb4f3fa15f8aeb3c74753310521a6c15/code/game/objects/items/tools/weldingtool.dm#L234-L259

use() logic needs to go into tool_use_check. reference the welder proc and adjust as needed please.

Reply:
Er, move which one into tool_use_check, the welder's tool_check_callback or plasma cutter's tool_start_check? If plasma cutter, they don't have continuous drain when on, or toggling, so they'll need an additional variable.

Reply:
you've broken the chain though. tool_use_check isn't called by anything but tool_start_check

Reply:
>The actual consumption of fuel/energy should be handled in use_tool

Then why does main definition of tool_check_callback have the comment of "don't override it"?  
And again, most of the code uses "amount = 0".
>tool_start_check is only intended to check if you're able to complete the action.

Which means it's not applicable to plasma cutters unless they take lump charge in the beginning, as opposed to simulating drain continuously (as I have them now). Tool_start_check does not pass valid arguments for amount of drain (majority of the code is calling with amount=0) or time (doesn't pass delay in any way whatsoever) so I _cannot_ use it to figure out how much charge to drain.
> This means everything in tool_use_check never gets checked.

tool_use_check is called by non-overridden tool_check_callback. 
https://github.com/tgstation/tgstation/blob/16dc86da24e5e139dcabcb70bf2c34b715519f6f/code/game/objects/items.dm#L768-L769
If the tool_check_callback is called, so is tool_use_check. Therefore any delayed action is going to call both.
But you're right in that it doesn't get called with _instant_ welding actions like turning plasma tiles to plasma (the chain for zero-delay welding actions with plasmacutter is `tool_start_check -> use  -> use_tool -> return;`, for weldingtool, replace use with tool_use_check)
>The chain in this situation would be

use_tool only calls tool_start_check if "delay" arg is 0. Your examples have 5 and 20 respectively. The actual chain is `tool_start_check -> use -> return; use_tool -> (tool_check_callback -> tool_use_check)*[delay] -> return;`
And tool_use_check only calls use() every 10 ticks. It drains the amounts I specified in the commit, because I've tested that.